python/cudf/cudf/core/frame.py

```diff
@@ -304,6 +304,70 @@ def _unaryop(self, op):
             result._data[name] = col.unary_operator(op)
         return result
 
+    def interleave_columns(self):
+        """
+       Interleave Series columns of a table into a single column.
+
+        Converts the column major table `cols` into a row major column.
+        Parameters
+        ----------
+        cols : input Table containing columns to interleave.
+
+        Example
+        -------
+        >>> df = DataFrame([['A1', 'A2', 'A3'], ['B1', 'B2', 'B3']])
+        >>> df
+        0    [A1, A2, A3]
+        1    [B1, B2, B3]
+        >>> df.interleave_columns()
+        0    A1
+        1    B1
+        2    A2
+        3    B2
+        4    A3
+        5    B3
+
+        Returns
+        -------
+        The interleaved columns as a single column
+        """
+
+        return self._constructor_sliced(
```

Comment:
in the case of `MultiIndex` this will return `Index` which I don't think can currently trivially construct from a `Column`. We'd either need to fix that or handle this in a different way.